src/binary-writer-spec.cc

```diff
@@ -160,6 +162,38 @@ void BinaryWriterSpec::WriteTypeObject(Type type) {
   json_stream_->Writef("}");
 }
 
+void BinaryWriterSpec::WriteF32(uint32_t f32_bits, ExpectedNan expected) {
+  switch (expected) {
+    case ExpectedNan::None:
+      json_stream_->Writef("\"%u\"", f32_bits);
+      break;
+
+    case ExpectedNan::Arithmetic:
+      WriteString("nan:arithmetic");
```

Comment:
Unrelated: Why is this called BinaryWriter?

Reply:
Good question, it should be JSONWriter I suppose. I think it was historical, because the old binary-writer used to contain this stuff too, so I probably kept the name when I split it out.